src/util/newconfig/ConfigFileJson.cpp

```diff
@@ -165,4 +181,97 @@ ConfigFileJson::flattenJson(boost::json::object const& obj, std::string const& p
     }
 }
 
+/*
+{
+    array: [
+        { a: 1},
+        { b: 2}
+    ]
+}
+
+array.[].a
+array.[].b
+
+*/
+
+void
+ConfigFileJson::flattenJsonNonRecursive(boost::json::object const& jsonRootObject)
```

Comment:
nit: If we don't need the original implementation anymore, perhaps this can be renamed to `flattenJson` 🔢